src/Analyzers/Core/Analyzers/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -104,19 +107,33 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                     }
                 }
 
-                static bool IsCandidateField(IFieldSymbol symbol) =>
+                static bool IsCandidateField(IFieldSymbol symbol, INamedTypeSymbol threadStaticAttribute) =>
                         symbol.DeclaredAccessibility == Accessibility.Private &&
                         !symbol.IsReadOnly &&
                         !symbol.IsConst &&
                         !symbol.IsImplicitlyDeclared &&
                         symbol.Locations.Length == 1 &&
                         symbol.Type.IsMutableValueType() == false &&
-                        !symbol.IsFixedSizeBuffer;
+                        !symbol.IsFixedSizeBuffer &&
+                        !ContainsAttribute(symbol, threadStaticAttribute);
```

Comment:
💡 This can also be written with a `static` lambda. If you don't use this approach, it would make sense to at least simplify the `ContainsAttribute` method to use the `Any` overload that takes a state argument.

```suggestion
                        !symbol.GetAttributes().Any(static (a, threadStaticAttribute) => SymbolEqualityComparer.Default.Equals(a.AttributeClass, threadStaticAttribute), threadStaticAttribute);
```

Reply:
I'm not currently on the machine I use for roslyn to confirm, but I can't see this `Any` overload in the docs site

https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=net-5.0

Reply:
I think Roslyn may define it internally